common/utils.h

```diff
@@ -158,4 +158,63 @@ int tmpdir_mkstemp(const tal_t *ctx, const char *template TAKES, char **created)
  */
 char *str_lowering(const void *ctx, const char *string TAKES);
 
+/**
+ * str_expand: substitute variable references in string via callback function.
+ * @ctx: the tal context from which to allocate the returned string.
+ * @str: the string containing variable references as described below.
+ * @subst: a callback function to resolve variable names to values.
+ * @ptr: an opaque pointer to be passed as the first parameter to @subst.
```

Comment:
Sorry, this function strikes me as an exploit waiting to happen.  Let's just get rid of renepay/debug...

Reply:
Nobody ever likes my hardcore string manipulation code. 🤷🏻‍♂️

Whatever. As long as the tests aren't hard-coded to write to `/tmp`, I'll be happy.